### PR TITLE
feat: reinit project with Xcode 12

### PR DESCRIPTION
**Pull request by kuoruan**
* Clean config files
* New project structure
* Update info.plist
* Update build version

### CODE REVIEW

**Comment by kuoruan** [comment]
image

image

**Comment by kuoruan** [comment]
We should always update the version numbers in `Info.plist`
First, the `Version number` and `Build number` will auto sync to `Squirrel.xcodeproj`
Second, we can update the version number and marketing number with `xcrun agvtool`

As Sparkle say, we'd better set the build number to a machine-readable version, an integer is preferred.

https://sparkle-project.org/documentation/publishing/#internal-build-numbers

**Review by lotem — COMMENTED** [review]
Can we add Sparkle.framework in a separate PR?

**Comment by lotem** [comment]
The project upgrade breaks some scripts. Need more time to test the affected workflows, including update from old clients after the version number change.

Let's add `Sparkle.framework` first if it doesn't depend on the project upgrade.

File: Squirrel/Info.plist [hunk 1]
```diff
@@ -16,10 +16,12 @@
 	<string>Squirrel</string>
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
+	<key>CFBundleShortVersionString</key>
+	<string>0.14.0</string>
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
-	<string>$(CURRENT_PROJECT_VERSION)</string>
+	<string>32</string>
```
[the remark this answers is not in the document]
Comment (kuoruan) [reply Squirrel/Info.plist:24]: The tags number of this project, 32 tags

**Comment by lotem** [comment]
I'm working on cherry-picking some of the comments in this PR.

**Comment by lotem** [comment]
Hi, I updated the project to Xcode 10 compatible version. That helps users to build the project (non-universal) with the minimal required version of Xcode - that currently being Xcode 10 with macOS 10.14 SDK.

**Pull request closed without merging**